packages/teleport/src/Apps/AppList/AppList.tsx

```diff
@@ -1,106 +1,170 @@
-import { NewTab } from 'design/Icon';
+import InputSearch from 'teleport/components/InputSearch';
 
-export default function AppList({ apps = [] }: Props) {
-  const $apps = apps.map(app => <Item mb={4} mr={5} app={app} key={app.id} />);
-  return <Flex flexWrap="wrap">{$apps}</Flex>;
-}
+function AppListt(props: Props) {
+  const { apps = [], pageSize = 20 } = props;
+
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    name: SortTypes.DESC,
+  });
+
+  const [searchValue, setSearchValue] = useState('');
+
+  function sortAndFilter(search) {
+    const filtered = apps.filter(obj =>
+      isMatch(obj, search, {
+        searchableProps: ['name', 'address', 'labels'],
+        cb: searchAndFilterCb,
+      })
+    );
+
+    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
+    const sorted = sortBy(filtered, columnKey);
+    if (sortDir[columnKey] === SortTypes.ASC) {
+      return sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  function onSortChange(columnKey: string, sortDir: string) {
+    setSortDir({ [columnKey]: sortDir });
+  }
+
+  const data = sortAndFilter(searchValue);
 
-function Item(props: ItemProps) {
-  const { app, ...rest } = props;
   return (
-    <StyledAppListItem
-      width="240px"
-      height="240px"
-      borderRadius="3"
-      flexDirection="column"
-      alignItems="center"
-      justifyContent="center"
-      bg="primary.light"
-      {...rest}
-    >
-      <Flex width="100%" justifyContent="center">
-        <MenuIcon buttonIconProps={menuActionProps}>
-          <MenuItem as="a" href={app.launchUrl} target="_blank">
-            Open
-          </MenuItem>
-        </MenuIcon>
+    <>
+      <Flex flex="0 0 auto" mb={4}>
+        <InputSearch
+          mr="3"
+          onChange={(e: React.SetStateAction<string>) => setSearchValue(e)}
+        />
       </Flex>
-      <Flex
-        flex="1"
-        alignItems="center"
-        justifyContent="center"
-        flexDirection="column"
+      <StyledTable pageSize={pageSize} data={data}>
+        <Column
+          columnKey="name"
+          header={
+            <SortHeaderCell
+              sortDir={sortDir.name}
+              onSortChange={onSortChange}
+              title="Name"
+            />
+          }
+          cell={<TextCell />}
+        />
+        <Column
+          columnKey="address"
+          header={
+            <SortHeaderCell
+              sortDir={sortDir.address}
+              onSortChange={onSortChange}
+              title="Address"
+            />
+          }
+          cell={<AddressCell />}
+        />
+        <Column header={<Cell>Labels</Cell>} cell={<LabelCell />} />
+        <Column header={<Cell />} cell={<ActionCell />} />
+      </StyledTable>
+    </>
+  );
+}
+
+const AddressCell = props => {
+  const { rowIndex, data } = props;
+  const { address } = data[rowIndex];
+
+  return (
+    <Cell>
+      <Text as="a" href={address} target="_blank" color="text.primary">
```

Comment:
i think `href` needs to be `luanchUrl` @alex-kovoy ?

Reply:
@kimlisa  it will redirect to the laucher automatically if not logged-in, so lets use the actual address

Reply:
the `open` button is also using the `launchUrl`

Reply:
@amha19 thanks for double checking, now as I am recalling it, we should be using `launchUrl` because `launchUrl` has more precise application parameters (due to some internal details of how it's been implemented on the backend). 

Okay, lets _not make_ this URL clickable, just display this full URL as text. There needs to be 1 way of opening an application and this is via "OPEN" button.